packages/expo-ui/ios/Modifiers/ViewModifierRegistry.swift

```diff
@@ -861,27 +1037,99 @@ private func parseAlignment(_ alignmentString: String) -> Alignment {
 
 internal extension Color {
   init(hex: String) {
-    let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
-    var int: UInt64 = 0
-    Scanner(string: hex).scanHexInt64(&int)
-    let a, r, g, b: UInt64
-    switch hex.count {
-    case 3: // RGB (12-bit)
-      (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
-    case 6: // RGB (24-bit)
-      (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
-    case 8: // ARGB (32-bit)
-      (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
-    default:
-      (a, r, g, b) = (1, 1, 1, 0)
+    let input = hex.trimmingCharacters(in: .whitespacesAndNewlines)
+
+    // Check if it's a hex color (starts with # or is pure hex)
+    if input.hasPrefix("#") || input.allSatisfy({ $0.isHexDigit }) {
+      let hex = input.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
+      var int: UInt64 = 0
+      Scanner(string: hex).scanHexInt64(&int)
+      let a, r, g, b: UInt64
+      switch hex.count {
+      case 3: // RGB (12-bit)
+        (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
+      case 6: // RGB (24-bit)
+        (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
+      case 8: // ARGB (32-bit)
+        (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
+      default:
+        (a, r, g, b) = (1, 1, 1, 0)
+      }
+
+      self.init(
+        .sRGB,
+        red: Double(r) / 255,
+        green: Double(g) / 255,
+        blue: Double(b) / 255,
+        opacity: Double(a) / 255
+      )
+    } else {
+      // Handle basic named colors
+      // Try system colors first
+      if let systemColor = uiColorFromNamedColor(input.lowercased()) {
+        self.init(systemColor)
+      } else {
+        // Fallback to clear color if no match
+        self = .clear
+      }
     }
+  }
+}
 
-    self.init(
-      .sRGB,
-      red: Double(r) / 255,
-      green: Double(g) / 255,
-      blue: Double(b) / 255,
-      opacity: Double(a) / 255
-    )
+private extension Character {
+  var isHexDigit: Bool {
+    return self.isNumber || ("a"..."f").contains(self) || ("A"..."F").contains(self)
+  }
+}
+
+private func uiColorFromNamedColor(_ name: String) -> UIColor? {
```

Comment:
We have color conversions in e-m-c. Maybe we can make them available here and avoid needing this?

Reply:
No it works with string, hex, named colors, swift ui named colors

https://github.com/user-attachments/assets/b69f1365-451e-4d81-ac6c-7279ff01d931